### DISCUSSION TITLE
Introducing a solution when the download fails when executing the "scoop update" command

### DISCUSSION THREAD

**Discussion by bianyun** [ScoopInstaller/Scoop#6237] [General]
When executing command 'scoop update', the downloading of app files may fail, but actually you can successfully download it individually through browser (Chrome). This happens mostly because of the GFW blocking you access some websites freely (such as github.com, etc.). 

If you can access github.com with the help of vpn, then you can download the apps individually by chrome (or other browsers). The url of the scoop app can be found in the app_bucket.json (SCOOP_HOME/buckets/<bucket_name>/bucket/<app_bucket_name>.json).

After you download the app file through the url, then you should rename it by conforming the filename pattern of scoop, and move it to scoop cache dir. Then execute 'scoop update' command, and everything is going ok.

The last thing is what name should the app file be renamed to? According to this pull request (https://github.com/ScoopInstaller/Scoop/pull/5929), the cache file name pattern was changed in this commit (https://github.com/ScoopInstaller/Scoop/pull/5929/commits/13b1f0266e73860e587162532861d501990dde20), and the old name pattern support will be remove soon. 

![Snipaste_2024-12-02_11-39-04](https://github.com/user-attachments/assets/1490e0f4-72df-46a5-8d15-efe948c8d757)

So you should rename the download app file to the name conforming new pattern, which is 
`<app_name>#<version>#<first_7_hex_char_of_sha_256_hash_of_url>.<file_extension>`
such as:
```
dbeaver#24.3.0#c3ec014.zip
gpg4win#4.4.0#407cb9f.exe
```

**Comment by HUMORCE** [comment]
Why not applying proxy servers to Scoop as well? `scoop config -h`

If you just want to know the cached filename:

```
❯ scoop download 7zip
INFO  Downloading '7zip' [64bit] from main bucket

### Holding CTRL + C

~ took 3s
❯ ^C
~ took 3s
❯ ^C
~ took 3s
❯ ^C
~ took 3s
❯ ls .\scoop\cache\

    Directory: C:\Users\humorce\scoop\cache

Mode                 LastWriteTime         Length Name
----                 -------------         ------ ----
-a---           5/31/2025  2:24 AM              0 7zip#24.09#569098e.msi.download
```
